### ISSUE TITLE
Multiple + Array entity + Ajax problem

### ISSUE THREAD

**Issue by jadrovski** [kartik-v/yii2-widget-select2#7]
Got error "Array to string conversion" from Yii2 framework when using like so:
```php
            <?= $form->field($model, 'hints')->widget(SelectAjaxWrapper::className(), [
                'url' => Url::to(['entity-selector/hints']),
                'placeholder' => 'Выберите подсказку ...',
                'options' => [
                    'multiple' => true
                ],
            ]); ?>
```

I have wrapped this widget:

```php
class SelectAjaxWrapper extends Select2
{
    public $url;
    public $placeholder;

    public function init()
    {

        $this->pluginOptions['minimumInputLength'] = 3;
        $this->pluginOptions['allowClear'] = true;
        $this->pluginOptions['ajax']['url'] = $this->url;
        $this->pluginOptions['ajax']['dataType'] = 'json';
        $this->pluginOptions['ajax']['data'] = new JsExpression('function(term,page){return{search:term};}');
        $this->pluginOptions['ajax']['results'] = new JsExpression('function(data,page){return{results:data.results};}');
        $this->pluginOptions['initSelection'] = new JsExpression('function (element, callback) {
    var id = $(element).val();
    if (id !== "") {
        $.ajax("{$this->url}?id=" + id, {
            dataType: "json"
        }).done(function (data) {
            callback(data.results);
        });
    }
}');
        $this->options['placeholder'] = $this->placeholder;
        parent::init();
    }
}
```

It's working when I using simple data array and no ajax.
Please help :)

**Comment by kartik-v** [comment]
As the message says, you seem to be returning an array where a string is expected. Most likely you may need to look at your AJAX action that generates the list... it must be JSON encoded - you may be returning an array. Similarly you may debug/check other aspects in the code where this could be happening.

Check the [Select2 widget ajax demo section](http://demos.krajee.com/widget-details/select2#ajax) for examples.

**Comment by Erkebulan69** [comment]
I have the same error, when I remove 'multiple' options, it works just fine for one selection, but not for multiple.
I checked generated list, it is JSON

**Comment by sebako86** [comment]
Hey, got the same Problem when using select2 in a grid as Filter. I can search and select a value, I see the tag beeing added to the input but than I get an exception from yii2 BaseHtml line 513:


* See [[renderTagAttributes()]] for details on how attributes are being rendered.
     * @return string the generated input tag
     */
    public static function input($type, $name = null, $value = null, $options = [])
    {
        if (!isset($options['type'])) {
            $options['type'] = $type;
        }
        $options['name'] = $name;
        $options['value'] = $value === null ? null : (string) $value;
        return static::tag('input', '', $options);
    }
 
    /**
     * Generates an input button.
     * @param string $label the value attribute. If it is null, the value attribute will not be generated.
     * @param array $options the tag options in terms of name-value pairs. These will be rendered as
     * the attributes of the resulting tag. The values will be HTML-encoded using [[encode()]].
     * If a value is null, the corresponding attribute will not be rendered.

**Comment by sebako86** [comment]
The Code I have used in gridview:

`[
		    'attribute'=>'geos',
		    'vAlign'=>'middle',
		    'width'=>'210px',
		    'value'=>function ($model, $key, $index, $widget) { 
	        	return $model->getOfferCountries();
	    	},
	    	'filterType'=>GridView::FILTER_SELECT2,
            'filterWidgetOptions'=>[
                'pluginOptions'=>[
        			'minimumInputLength' => 2,
                	'allowClear'=>false,
                	'tags'=>true,
                	'multiple'=>true,
                	'tokenSeparators' => [',', ' '],
			        'templateResult' => new JsExpression('format'),
			        'templateSelection' => new JsExpression('format'),
	                'ajax' => [
			            'url' => Url::to(['offers/geos']),
			            'dataType' => 'json',
			            'data' => new JsExpression('function(params) { return {q:params.term}; }')
			        ],
		        ],
            ],
	        'filterInputOptions' => ['placeholder' => 'Geo'],
	        'format'=>'raw'
		],`
